pkg/webhooks/sgx/sgx.go

```diff
@@ -63,19 +99,71 @@ func (s *SgxMutator) Handle(ctx context.Context, req admission.Request) admissio
 		if epcSize, ok := requestedResources[epc]; ok {
 			totalEpc += epcSize
 
-			attestation, found := pod.Annotations[provisionAnnotation]
-			if found && attestation == "yes" {
+			// Quote Generation Modes:
+
+			//
+			// in-process: A container has its own quote provider library library: In this mode,
+			// the container needs a handle to /dev/sgx/provision (sgx.intel.com/provision resource).
+			// out-of-process: A container uses Intel aesmd. In this mode, the container must talk to
+			// aesmd over /var/run/aesmd/aesm.sock. aesmd can run either as a side-car or a DaemonSet
+			//
+			// Mode selection: The mode selection is done by setting sgx.intel.com/quote-provider annotation
+			// to a value that specifies the container name. If the annotation matches the container requesting
+			// SGX EPC resources, the webhook adds both /dev/sgx/provision and /dev/sgx/enclave resource requests.
+			// Without sgx.intel.com/quote-provider annotation set, the container is not able to generate quotes
+			// for its enclaves. When pods set sgx.intel.com/quote-provider: "aesmd", Intel aesmd specific volume
+			// mounts are added. In both DaemonSet and sidecar deployment scenarios for aesmd, its container name
+			// must be set to "aesmd" (TODO: make it configurable?).
+
+			if quoteProvider == container.Name {
 				pod.Spec.Containers[idx].Resources.Limits[corev1.ResourceName(provision)] = resource.MustParse("1")
 				pod.Spec.Containers[idx].Resources.Requests[corev1.ResourceName(provision)] = resource.MustParse("1")
 			}
+
 			pod.Spec.Containers[idx].Resources.Limits[corev1.ResourceName(encl)] = resource.MustParse("1")
 			pod.Spec.Containers[idx].Resources.Requests[corev1.ResourceName(encl)] = resource.MustParse("1")
+
+			switch quoteProvider {
+			// mutate logic for Intel aesmd users
+			case "aesmd":
+				if container.VolumeMounts == nil {
+					container.VolumeMounts = make([]corev1.VolumeMount, 0)
+				}
+				if container.Env == nil {
+					container.Env = make([]corev1.EnvVar, 0)
+				}
+
+				container.VolumeMounts = append(container.VolumeMounts,
+					corev1.VolumeMount{
+						Name:      "aesmd-socket",
+						MountPath: "/var/run/aesmd",
+					})
+
+				// count how many containers need aesmd mounts
+				aesmdMounts++
+
+				// this sets SGX_AESM_ADDR for aesmd itself but it's harmless
+				container.Env = append(container.Env,
+					corev1.EnvVar{
+						Name:  "SGX_AESM_ADDR",
+						Value: "1",
+					})
+			// case xyz: the mutate logic for other out-of-process quote providers
+			default:
```

Comment:
Should maybe log an error and tell "aesmd" is the only supported value. :-)

Reply:
dropped these for now